### PR TITLE
k40-whisperer: init at 0.59

### PR DESCRIPTION
**Pull request by fooker**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/k40-whisperer/default.nix [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  buildPhase = "";
+
+  installPhase = ''
+    mkdir -p $out/bin
+    mkdir -p $out/lib/udev/rules.d
+
+    ln -s ${udevRule} $out/lib/udev/rules.d/97-k40-whisperer.rules
```
Comment (mweinelt) [inline pkgs/applications/misc/k40-whisperer/default.nix:40]: `programs.k40-whisperer.enable` when? :wink:

File: pkgs/applications/misc/k40-whisperer/default.nix [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  pythonEnv = python3.withPackages (ps: with ps; [
+    lxml
+    pyusb
+    pillow
+    pyclipper
+    tkinter
+  ]);
+
+  udevRule = writeText "k40-whisperer.rules" ''
+    SUBSYSTEM=="usb", ATTRS{idVendor}=="1a86", ATTRS{idProduct}=="5512", ENV{DEVTYPE}=="usb_device", MODE="0664", GROUP="users"
```
Comment (mweinelt) [inline pkgs/applications/misc/k40-whisperer/default.nix:19]: Would it be prudent to create a dedicated group for this device? If so, we could also use `programs.k40-whisperer.users`.

**Comment by fooker** [comment]
Just realized that there was a hard-coded dependency to inkscape which needed patching. Updated the PR accordingly.

File: pkgs/applications/misc/k40-whisperer/default.nix (hunk 2 above)
Comment (fooker) [reply pkgs/applications/misc/k40-whisperer/default.nix:20]: Should a package depend on a config module?

Comment (fooker) [reply pkgs/applications/misc/k40-whisperer/default.nix]: Configurable via argument, defaulting to `k40`.

File: pkgs/applications/misc/k40-whisperer/default.nix (hunk 1 above)
Comment (fooker) [reply pkgs/applications/misc/k40-whisperer/default.nix:50]: Added a module with `enable`, `group` and `package`.

**Comment by fooker** [comment]
Entry in release-notes added.

**Pull request merged**